src/sentry/static/sentry/app/components/hook.jsx

```diff
@@ -0,0 +1,60 @@
+import PropTypes from 'prop-types';
+import Reflux from 'reflux';
+import createReactClass from 'create-react-class';
+
+import HookStore from 'app/stores/hookStore';
+
+/**
+ * Instead of accessing the HookStore directly, use this.
+ *
+ * If the hook slot needs to perform anything w/ the hooks, you can pass a function as a child and you will receive an object with a `hooks` key
+ *
+ * example:
+ *
+ * <Hook name="my-hook">
+ * {({hooks}) => hooks.map(hook => (
+ *  <Wrapper>
+ *    {hook}
+ * </Wrapper>
+ * ))}
+ * </Hook>
+ *
+ */
+const Hook = createReactClass({
+  displayName: 'Hook',
+  propTypes: {
+    name: PropTypes.string.isRequired,
+  },
+  mixins: [Reflux.listenTo(HookStore, 'handleHooks')],
+
+  getInitialState() {
+    let {name, ...props} = this.props;
+
+    return {
+      hooks: HookStore.get(name).map(cb => cb(props)),
+    };
+  },
+
+  handleHooks(name, hooks) {
+    // Make sure that the incoming hook update matches this component's hook name
+    if (name !== this.props.name) return;
+
+    this.setState(state => ({
+      hooks: hooks.map(cb => cb(this.props)),
```

Comment:
Probably should destructure `name` from this to be consistent with `getInitialState`

Reply:
agreed